### PR TITLE
docs: add information about the build system

### PR DESCRIPTION
**Pull request by Okeanos**
As requested by @bgilbert during a [similar PR](https://github.com/coreos/butane/pull/298) for butane.

### CODE REVIEW

File: docs/development.md [hunk 1]
```diff
@@ -108,6 +108,13 @@ UUIDs may be required in the following fields of a Test object: In, Out, and Con
 
 Create a new [release checklist](https://github.com/coreos/ignition/issues/new?labels=kind/release&template=release-checklist.md) and follow the steps there.
 
+## The build process
+
+Note that the `build` script included in this repository is a convenience script only and not used for the actual release binaries. Those are built using a `ignition.spec` maintained in [Fedora rpms/ignition](https://src.fedoraproject.org/rpms/ignition).
```
Comment (bgilbert) [inline docs/development.md:113]: The published ignition-validate container isn't built from the specfile; the Dockerfile uses `./build_for_container`.

nit: "an `ignition.spec`"

**Review by bgilbert — COMMENTED** [review]
Thanks!

File: docs/development.md (hunk 1 above)
Comment (Okeanos) [reply docs/development.md]: Should be fixed now

File: docs/development.md [hunk 2]
```diff
@@ -108,6 +108,15 @@ UUIDs may be required in the following fields of a Test object: In, Out, and Con
 
 Create a new [release checklist](https://github.com/coreos/ignition/issues/new?labels=kind/release&template=release-checklist.md) and follow the steps there.
 
+## The build process
+
+Note that the `build` script included in this repository is a convenience script only and not used for the actual release binaries. Those are built using an `ignition.spec` maintained in [Fedora rpms/ignition](https://src.fedoraproject.org/rpms/ignition) and the `build_for_container` script in case of the ignition-validate container, respectively.
```
Comment (bgilbert) [inline docs/development.md:113]: Could we make it clearer that the rest of this section covers `ignition.spec` and not `build_for_container`?  Maybe something like:

```suggestion
Note that the `build` script included in this repository is a convenience script only and not used for the actual release binaries. Those are built using an `ignition.spec` maintained in [Fedora rpms/ignition](https://src.fedoraproject.org/rpms/ignition). (The `ignition-validate` [container](https://quay.io/repository/coreos/ignition-validate) is built by the `build_for_container` script, which is not further described here.)
```

Comment (Okeanos) [reply docs/development.md]: Applied your changes :)

**Review by bgilbert — APPROVED** [review]
Thanks!

**Pull request merged**